src/vcSceneLayer.cpp

```diff
@@ -591,28 +601,41 @@ udResult vcSceneLayer_LoadTextureData(const vcSceneLayer *pSceneLayer, vcSceneLa
     // TODO: (EVC-542) other formats (this information is in sharedResource.json)
     for (size_t f = 0; f < udLengthOf(pExtensions); ++f)
     {
-      udSprintf(&pPathBuffer, "zip://%s%c%s.%s", pNode->pURL, pSceneLayer->pathSeparatorChar, pNode->pTextureData[i].pURL, pExtensions[f]);
-      if (udFile_Load(pPathBuffer, (void**)&pFileData, &fileLen) != udR_Success)
+      udSprintf(&pPathBuffer, "%s%c%s.%s", pNode->pURL, pSceneLayer->pathSeparatorChar, pNode->pTextureData[i].pURL, pExtensions[f]);
+
+      if (udFile_SetSubFilename(pSceneLayer->pFile, pPathBuffer, &fileLen) != udR_Success)
         continue;
 
-      pPixelData = (uint8_t*)stbi_load_from_memory((stbi_uc*)pFileData, (int)fileLen, (int*)&width, (int*)&height, (int*)&channelCount, 4);
-      
+      pFileData = udAllocType(char, fileLen, udAF_None);
+      UD_ERROR_NULL(pFileData, udR_MemoryAllocationFailure);
+        
+      if (udFile_Read(pSceneLayer->pFile, pFileData, fileLen) != udR_Success)
+      {
+        udFree(pFileData);
+        continue;
+      }
+
+      pPixelData = (uint8_t *)stbi_load_from_memory((stbi_uc *)pFileData, (int)fileLen, (int *)&width, (int *)&height, (int *)&channelCount, 4);
+
       if (pPixelData != nullptr)
       {
         pNode->pTextureData[i].width = width;
         pNode->pTextureData[i].height = height;
         pNode->pTextureData[i].pData = (uint8_t *)udMemDup(pPixelData, sizeof(uint32_t) * width * height, 0, udAF_None);
       }
 
-      udFree(pFileData);
       stbi_image_free(pPixelData);
+      udFree(pFileData);
       result = udR_Success;
       break;
     }
   }
 
 epilogue:
   udFree(pPathBuffer);
+  udFree(pFileData);
```

Comment:
Might be no actual codepath to this that dosen't already free pFileData

Reply:
You are right. Just for future-proofing though I'm going to leave it in.